source/grid/tria_accessor.cc

```diff
@@ -1329,139 +1329,53 @@ namespace
-                     std::sqrt(
-                       a + 0.44888729929169013 * b + 0.0048207809894260144 * c +
-                       1.3399810435848563 * d + 0.13886368840594743 * e +
-                       0.09303735505312187 * f) +
-                   0.10632332575267359 * std::sqrt(a + 0.44888729929169013 * b +
-                                                   0.10890625570683385 * c +
-                                                   1.3399810435848563 * d +
-                                                   0.66001895641514374 * e +
-                                                   0.44220644500147605 * f) +
-                   0.10632332575267359 *
-                     std::sqrt(a + 0.44888729929169013 * b +
-                               0.44888729929169013 * c +
-                               1.3399810435848563 * d + 1.3399810435848563 * e +
-                               0.89777459858338027 * f) +
-                   0.056712962962962937 *
-                     std::sqrt(a + 0.44888729929169013 * b +
-                               0.86595709258347853 * c +
-                               1.3399810435848563 * d + 1.8611363115940525 * e +
-                               1.2469436885317342 * f) +
-                   0.03025074832140047 * std::sqrt(a + 0.86595709258347853 * b +
-                                                   0.0048207809894260144 * c +
-                                                   1.8611363115940525 * d +
-                                                   0.13886368840594743 * e +
-                                                   0.12922212642709538 * f) +
-                   0.056712962962962937 *
-                     std::sqrt(
-                       a + 0.86595709258347853 * b + 0.10890625570683385 * c +
-                       1.8611363115940525 * d + 0.66001895641514374 * e +
-                       0.61419262306231814 * f) +
-                   0.056712962962962937 *
-                     std::sqrt(a + 0.86595709258347853 * b +
-                               0.44888729929169013 * c +
-                               1.8611363115940525 * d + 1.3399810435848563 * e +
-                               1.2469436885317342 * f) +
-                   0.03025074832140047 *
-                     std::sqrt(a + 0.86595709258347853 * b +
-                               0.86595709258347853 * c +
-                               1.8611363115940525 * d + 1.8611363115940525 * e +
-                               1.7319141851669571 * f);
-          }
-
-        // the face is planar. then its area is 1/2 of the norm of the
-        // cross product of the two diagonals
-        const Tensor<1, 3> v12        = accessor.vertex(2) - accessor.vertex(1);
-        const Tensor<1, 3> twice_area = cross_product_3d(v03, v12);
-        return 0.5 * twice_area.norm();
+        const Point<3> x0 = accessor.vertex(0), x1 = accessor.vertex(1),
+                       x2 = accessor.vertex(2), x3 = accessor.vertex(3);
```

Comment:
```suggestion
        const Point<3> &x0 = accessor.vertex(0);
        const Point<3> &x1 = accessor.vertex(1);
        const Point<3> &x2 = accessor.vertex(2);
        const Point<3> &x3 = accessor.vertex(3);
```
I believe they can all be references. And I think we avoid multiple definition per line elsewhere in the library. So I suggest to write it like this :)

Reply:
I agree with using different lines, but from a performance point of view I would still prefer copies. Since we do not call other functions (in release mode), the compiler will try to keep data in registers as much as possible without spilling to L1 cache, and at the same time the copy indicates that there can not be any aliasing.